tests/lightning_tensor/test_expval.py

```diff
@@ -0,0 +1,398 @@
+        [
+            (
+                [qml.PauliX(wires=[0]), qml.PauliX(wires=[1])],
+                qml.RY,
+                lambda theta, phi: np.array([np.sin(theta) * np.sin(phi), np.sin(phi)]),
+            ),
+            (
+                [qml.PauliY(wires=[0]), qml.PauliY(wires=[1])],
+                qml.RX,
+                lambda theta, phi: np.array([0, -np.cos(theta) * np.sin(phi)]),
+            ),
+            (
+                [qml.PauliZ(wires=[0]), qml.PauliZ(wires=[1])],
+                qml.RX,
+                lambda theta, phi: np.array([np.cos(theta), np.cos(theta) * np.cos(phi)]),
+            ),
+            (
+                [qml.Hadamard(wires=[0]), qml.Hadamard(wires=[1])],
+                qml.RY,
+                lambda theta, phi: np.array(
+                    [
+                        np.sin(theta) * np.sin(phi) + np.cos(theta),
+                        np.cos(theta) * np.cos(phi) + np.sin(phi),
+                    ]
+                )
+                / np.sqrt(2),
+            ),
+        ],
+    )
+    def test_single_wire_observables_expectation(self, Obs, Op, expected_fn, theta, phi, tol, dev):
+        """Test that expectation values for single wire observables are correct"""
+
+        tape = qml.tape.QuantumScript(
+            [Op(theta, wires=[0]), Op(phi, wires=[1]), qml.CNOT(wires=[0, 1])],
+            [qml.expval(Obs[0]), qml.expval(Obs[1])],
+        )
+        result = execute(dev, tape)
+        expected = expected_fn(theta, phi)
+
+        assert np.allclose(result, expected, tol)
+
+    def test_hermitian_expectation(self, theta, phi, tol, dev):
+        """Tests an Hermitian operator."""
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(theta, wires=0)
+            qml.RX(phi, wires=1)
+            qml.RX(theta + phi, wires=2)
+
+            for idx in range(3):
+                qml.expval(qml.Hermitian([[1, 0], [0, -1]], wires=[idx]))
+
+        calculated_val = execute(dev, tape)
+        reference_val = calculate_reference(tape)
+
+        assert np.allclose(calculated_val, reference_val, tol)
+
+    def test_hamiltonian_expectation(self, theta, phi, tol, dev):
+        """Tests a Hamiltonian."""
+
+        ham = qml.Hamiltonian(
+            [1.0, 0.3, 0.3, 0.4],
+            [
+                qml.PauliX(0) @ qml.PauliX(1),
+                qml.PauliZ(0),
+                qml.PauliZ(1),
+                qml.PauliX(0) @ qml.PauliY(1),
+            ],
+        )
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(theta, wires=0)
+            qml.RX(phi, wires=1)
+            qml.RX(theta + phi, wires=2)
+
+            qml.expval(ham)
+
+        calculated_val = execute(dev, tape)
+        reference_val = calculate_reference(tape)
+
+        assert np.allclose(calculated_val, reference_val, tol)
+
+    def test_sparse_hamiltonian_expectation(self, theta, phi, tol, dev):
+        """Tests a Hamiltonian."""
+
+        ham = qml.SparseHamiltonian(
```

Comment:
Without sparse-MPO support, does it really make sense to test this?

Reply:
I don't think it is necessary. Technically, the full support for expectation values with sparse Hamiltonians is out of scope, even for the second epic. Still, I don't see the problem if we leave it here (it doesn't take a lot of time, and one test more is better than one test less)